src/mqtt_client.rs

```diff
@@ -436,6 +486,13 @@ where
                 Ok(())
             }
 
+            ReceivedPacket::PubAck(ack) => {
+                // No matter the status code the message is considered acknowledged at this point
+                self.session_state.handle_puback(ack.packet_identifier);
```

Comment:
We need to look at the ack ReasonCode (if present) as well to verify that the server gave us a good ACK

Reply:
But doesn't that imply that we still need to only handle the puback if reason_code >= 0x80 || reason_code == 0x00?

E.g. if server responds with reason_code 0x10, we shouldn't actually register the packet as sent, since there's no subscriber to actaully accept the message and we can't guarantee AtLeaseOnce

Reply:
Hmmm... I think I can get on-board with ignoring the reason code. I'm unsure what the corrective action would even be if we got an invalid reason code (hold onto it forever?). Let's move forward with ignoring the reason code for now then. If we need to, we can add logging of it in the future.